### PR TITLE
Adapt to coq/coq#17836 (sort poly)

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: src/Rewriter/Util/Tactics2/Constr.v
Comment (JasonGross) [inline src/Rewriter/Util/Tactics2/Constr.v:58]: Please version this file a la
https://github.com/mit-plv/rewriter/blob/26f5c8497a1e513b34ea65bcc6ec3b15b1b3d5a2/Makefile.local.common#L60
so that the change is backwards compatible.

File: src/Rewriter/Util/Tactics2/Proj.v
Comment (JasonGross) [inline src/Rewriter/Util/Tactics2/Proj.v:7]: Please version this file a la
https://github.com/mit-plv/rewriter/blob/26f5c8497a1e513b34ea65bcc6ec3b15b1b3d5a2/Makefile.local.common#L60
so that the change is backwards compatible.

**Review by JasonGross — CHANGES_REQUESTED** [review]

**Comment by SkySkimmer** [comment]
That's a lot of work for the overlay writer dealing with a build system I'm not familiar with. Can you do it yourself?

**Comment by JasonGross** [comment]
Sure, what's the timeline on getting the PR merged?  It'll probably take me a couple days at least to find time.

**Comment by SkySkimmer** [comment]
Probably not this week, I haven't finished the overlays and the doc, and then we will need reviewing and perf discussion etc

File: src/Rewriter/Util/plugins/inductive_from_elim.ml.v819
Comment (SkySkimmer) [inline src/Rewriter/Util/plugins/inductive_from_elim.ml.v819:1]: it doesn't pick use this file for some reason, instead it uses the v818 one

File: src/Rewriter/Util/Tactics2/Constr.v
Comment (JasonGross) [reply src/Rewriter/Util/Tactics2/Constr.v:58]: @SkySkimmer Could `iter_with_binders` be exposed in Ltac2 in Coq upstream?  Then we could just drop this definition in new versions of the file.

File: src/Rewriter/Util/plugins/inductive_from_elim.ml.v819
Comment (JasonGross) [reply src/Rewriter/Util/plugins/inductive_from_elim.ml.v819:1]: I'm on it, give me a bit and I'll split off the PR that adds a v819 suite of files

File: src/Rewriter/Util/Tactics2/Constr.v
Comment (JasonGross) [reply src/Rewriter/Util/Tactics2/Constr.v:58]: (`iter_with_full_binders` and `iter` would also be nice)

File: src/Rewriter/Util/plugins/inductive_from_elim.ml.v819
Comment (JasonGross) [reply src/Rewriter/Util/plugins/inductive_from_elim.ml.v819:1]: https://github.com/mit-plv/rewriter/pull/112 for the v819 files, about to go add versioning for the three relevant files in Tactics2

File: src/Rewriter/Language/Reify.v
Comment (JasonGross) [reply src/Rewriter/Language/Reify.v:425]: I've fixed this in #110

File: src/Rewriter/Rewriter/Reify.v
Comment (JasonGross) [reply src/Rewriter/Rewriter/Reify.v:404]: I've fixed this in #111

**Comment by JasonGross** [comment]
I hope #113 will pass CI and auto-merge in about 10--20 minutes.  #114 will take care of `Proj.v`, which is already compatible.  If you rebase this PR on top of that one, the changes should be pretty minimal (just the two v819 Tactics2 files).  Though if the constr iter functions can be upstreamed in Coq, that'd be even better

File: src/Rewriter/Util/Tactics2/Constr.v.v818
Comment (JasonGross) [inline src/Rewriter/Util/Tactics2/Constr.v.v818:87]: Sort poly isn't making it into 8.18, right?
```suggestion
    | Proj _p c => f n c
```

File: src/Rewriter/Util/Tactics2/DestProj.v.v818
Comment (JasonGross) [inline src/Rewriter/Util/Tactics2/DestProj.v.v818:9]:
```suggestion
  | Proj p v => (p, v)
```

File: src/Rewriter/Util/Tactics2/Proj.v.v816
Comment (JasonGross) [inline src/Rewriter/Util/Tactics2/Proj.v.v816:7]: This change has already been done for 8.19, this is the 8.16 file
```suggestion
  := let dummy := make (Rel (-1)) in
     Constr.equal (make (Proj x dummy)) (make (Proj y dummy)).
```

**Review by JasonGross — APPROVED** [review]
LGTM, happy to merge once the Coq PR lands and CI here passes

**Comment by ppedrot** [comment]
Please merge now.

**Pull request merged**
